src/pushsource/_impl/source.py

```diff
@@ -29,14 +31,42 @@ class SourceUrlError(ValueError):
 
 
 class SourceWrapper(object):
+    timeout = 900
+    poll_rate = 30
     # Internal class to ensure that all source instances support enter/exit
     # for with statements even if underlying instance doesn't implement it
     # (since it originally was not mandatory).
     def __init__(self, delegate):
         self.__delegate = delegate
 
     def __iter__(self):
-        return self.__delegate.__iter__()
+        """
+        Poll until a source file is present before yielding an item.
+
+        This is necessary due to a limitation, where file changes may not immediately propagate
+        to NFS.
+
+        Raises:
+            RuntimeError:
```

Comment:
Is it safe to offer an iterable which may crash mid-iteration and potentially leave the user with a partially completed operation on their side?

Reply:
Behavior changed